src/methods-and-traits/traits/implementing.md

```diff
@@ -0,0 +1,41 @@
+# Implementing Traits
+
+```rust,editable
+trait Pet {
+    fn talk(&self) -> String;
+
+    fn greet(&self) {
+        println!("Oh you're a cutie! What's your name? {}", self.talk());
+    }
+}
+
+struct Dog {
+    name: String,
+    age: i8,
+}
+
+impl Pet for Dog {
+    fn talk(&self) -> String {
+        format!("Woof, my name is {}!", self.name)
+    }
+}
+
+fn main() {
+    let fido = Dog { name: String::from("Fido"), age: 5 };
+    fido.greet();
+}
+```
+
+<details>
+
+- Types must implement the trait with an `impl <trait> for <type> { .. }` block.
```

Comment:
I'm not a fan of the mix between real syntax and pseudo-code, especially when Rust uses `<...>` for real things :smile: 

What about:

```suggestion
- To implement `Trait` for `Type`, you use an `impl Trait for Type { .. }` block.
```